flang/include/flang/Parser/parse-tree.h

```diff
@@ -5243,38 +5243,65 @@ struct AccEndBlockDirective {
 };
 
 // ACC END ATOMIC
-EMPTY_CLASS(AccEndAtomic);
+struct AccEndAtomic {
+  WRAPPER_CLASS_BOILERPLATE(AccEndAtomic, Verbatim);
+  CharBlock source;
+};
 
 // ACC ATOMIC READ
+struct AccAtomicReadDirective {
+  TUPLE_CLASS_BOILERPLATE(AccAtomicReadDirective);
+  std::tuple<Verbatim, AccClauseList> t;
+  CharBlock source;
```

Comment:
Both `Verbatim` and `AccClauseList` nodes have `source` data members.  Why do you need this one?